### DISCUSSION TITLE
Close button on electron build does not work

### DISCUSSION THREAD

**Discussion by dmm9** [eclipse-theia/theia#13937] [Q&A] [answered]
The close (X) button does not have any effect on electron build on Windows - using Theia 1.49.1, node 20, windows. The only way to quit is by killing the process in the task manager. Is this a bug or a wrong configuration on my side?

**Comment by msujew** [comment]
Hey @dmm9,

do you mean an electron build of your custom IDE? I cannot reproduce this using the example app in this repo or the Theia IDE. Can you reproduce the issue using the [Theia IDE app](https://theia-ide.org/#theiaidedownload)?

**Reply by dmm9** [reply]
Yes, I am building my custom IDE. The (X) works if I download the prebuilt installer or if I build the blueprint repo (latest version as of today)

By the way, the splash screen is smaller than the image, so a scroll bar appears. But that's another topic.
![grafik](https://github.com/user-attachments/assets/a3ab4503-f91a-477e-8c0a-cf84790ffbcf)

**Reply by dmm9** [reply]
@msujew Any ideas where can I look into? Unfortunately, I am not experienced with electron..

**Reply by msujew** [reply]
No idea, maybe you can provide a reproducible example? Sounds like a pretty unique problem. Maybe @tsmaeder knows more.

**Comment by tsmaeder** [comment, marked as answer]
The close button works on the electron example in main. I guess the simplest way would be to debug through the code that handles the close event from the electron `BrowserWindow` in `theia-electron-window.ts`. The "Launch Electron Back End" debug configuration from the theia repo should work (with changes for your concrete case) in your project as well.

**Reply by dmm9** [reply]
Thank you @tsmaeder . I have just revisited this issue (couldn't do this before for other reasons), and I have quickly found the mistake by creating (copying) the debug configuration. It has been straightforward.

I could see the debug message `Shutdown prevented by No reason given` when clicking on the (X) button, which was caused by a frontend contribution with `OnWillStop` always returning `true`.
